internal/cache/file/downloader/job.go

```diff
@@ -299,14 +303,46 @@ func (job *Job) downloadObjectAsync() {
 	}
 }
 
-// Download downloads object till the given offset if not already downloaded
-// and waits for download if waitForDownload is true.
-// The caller of this method should not read from cache if job status is FAILED.
-// ToDo (sethiay): Implement this function.
+// Download downloads object till the given offset and returns the status of
+// job. If the object is already downloaded or there was failure/cancellation in
+// download, then it returns the job status
 //
 // Acquires and releases LOCK(job.mu)
-func (job *Job) Download(ctx context.Context, offset int64, waitForDownload bool) (jobStatus JobStatus) {
+func (job *Job) Download(ctx context.Context, offset int64, waitForDownload bool) (jobStatus JobStatus, err error) {
```

Comment:
1. Expose a new API GetJobStatus.
2. Please add a new status INVALID.
3. Please document clearly for the client: for all status where client should not be advised to read from the downloaded local file.